test/end-to-end/test_docker_exporter.ps1

```diff
@@ -0,0 +1,135 @@
+# Utility function to create a custom tag for a container image that
+# is unique in this test run.
+function New-CustomTag {
+    $timestamp = Get-Date -UFormat "%Y%m%d%H%M%S"
+    "e2e-test-$timestamp"
+}
+
+# Export a core/redis container image.
+#
+# $extra_args are options for `hab pkg export docker` to influence
+# container creation. (e.g., pass "--multi-layer" to create a
+#multi-layer image.)
+function New-RedisImage() {
+    param(
+        [Parameter(Mandatory=$true)][string]$tag,
+        [Parameter(Mandatory=$false)][string]$extra_args
+    )
+    Write-Host (hab pkg export docker core/redis --tag-custom=$tag --base-pkgs-channel=$env:HAB_INTERNAL_BLDR_CHANNEL $extra_args | Out-String)
+    "core/redis:$tag"
+}
+
+# Run a given Habitat container image in the background, returning the
+# name of the container.
+#
+# $extra_args are for `docker run` and can affect how the container is
+# actually executed. (e.g., pass "--user=12354151" to see if the
+# Supervisor can run as a non-root user)
+function Start-Container() {
+    param(
+        [Parameter(Mandatory=$true)][string]$image,
+        [Parameter(Mandatory=$false)][string]$extra_args
+    )
+    $name="e2e-docker-export-container"
+    Write-Host (docker run -d --name=$name --rm --env=HAB_LICENSE=accept-no-persist $extra_args $image | Out-String)
+    "$name"
+}
+
+# If we can set and get a value from Redis running in the container,
+# then we know we created a container that can actually run.
+function Confirm-ContainerBehavior() {
```

Comment:
I think if we tested with nginx instead of redis and then changed this function to an http check of the default home page, we would have a test that would work on both linux and windows. We would just need to run the non-root describes inside of an `if($IsLinux)`

Reply:
Agreed... with luck I'll get around to this today.